### PR TITLE
Orientation lock support for Mini app

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
# Description
* Enable Orientation lock from the Mini app. Mini app can lock the orientation using the JS interfaces and the list of Orientation locks supported is,
    * Portrait
    * Landscape
* Updated sample app plist to support all orientations
* Updated sample app to enable Orientation Lock

## Links
[MINI-1751](https://jira.rakuten-it.com/jira/browse/MINI-1751)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptEnumerations.swift
Comment (corycaywood) [inline MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptEnumerations.swift:72]: Could this override the Host App's default orientation setting? For example, this scenario: 
1. The App's view controller is locked to portrait mode by default
2. Then the mini app locks to landscape mode
3. Then the mini app releases the lock
4. The SDK sets the view controller to `.all`

Would this cause the App's view controller to no longer be locked to portrait and enable auto-rotate? Is there a way to save the App's original orientation and set it when the lock is released?

File: Example/Controllers/DisplayController.swift
Comment (corycaywood) [inline Example/Controllers/DisplayController.swift:56]: This seems a little complicated that we have to ask the Host App to override the supported orientation interfaces for their view controller, navigation controller, and the avplayer. But I guess there isn't really any other way of doing this 😣

File: MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift
Comment (corycaywood) [inline MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift:188]: On Android, we have added a setting for `MiniAppMessageBridge.allowScreenOrientation`. So if the Host App doesn't enable screen orientation changes, then the SDK won't do anything when the mini app calls `setScreenOrientation`.

Does this make sense to add on iOS? If the Host App doesn't override `UIViewController. supportedInterfaceOrientations` will the SDK be able to change anything about the orientation?

**Review by corycaywood — COMMENTED** [review]
It's looking good 👍 See my comments below

File: Example/Controllers/DisplayController.swift
Comment (ts-leojoseph-r) [reply Example/Controllers/DisplayController.swift:56]: Yes, that's the sad part here. 

1. `override var supportedInterfaceOrientations: UIInterfaceOrientationMask`
 This is to override the supported orientation for the view controller i.e the display controller which the host app would use to display the mini-app. If the mini-app wants to lock the orientation for the viewcontroller(not just the video), they can do it only if the host app override this.
2. `public func navigationControllerSupportedInterfaceOrientations`
 DisplayController is pushed in the navigation controller and without this override, Viewcontroller is not locking the orientation. This is little strange for me too, when I tested in a small sample app, orientation lock is working fine without this override. I will verify it one more time.
3. `AVPlayerViewController - > override open var supportedInterfaceOrientations`
This is to override the orientation of the default media player. The host app can still skip this one, this is not mandatory. If the host app didn't override this, then the video wouldn't rotate automatically to landscape mode when the lock is set. User has to manually rotate the screen to lock in the landscape mode.

File: Example/Controllers/DisplayController.swift
Comment (Climbatize) [inline Example/Controllers/DisplayController.swift:3]: the use of AVKit is really annoying to me. I know it's really convenient and maybe the only solution, but it's an heavy framework to just manage orientation

**Review by Climbatize — APPROVED** [review]
If we are sure we can't find another way than using AVKit I approve this PR (after you check Cory's comments of course). It's a very smart way to bypass Apple restrictions on the subject by the way

File: Example/Controllers/DisplayController.swift
Comment (corycaywood) [reply Example/Controllers/DisplayController.swift:56]: Thanks, this is probably okay. About `AVPlayerViewController`, this is specifically what the orientation feature is needed for - to automatically set videos to landscape mode. 

Seems this will be confusing to the Host App, but it should be okay if we clearly document in the user guide how to handle orientation and what each override is for.

File: MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptEnumerations.swift
Comment (ts-leojoseph-r) [reply MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptEnumerations.swift]: This won't override the host app default orientation, returning `.all` would reset the host app to its orientation once the lock is released. However, it is not good to return `.all` without knowing the host app current orientation. I have updated the code now to return the `.landscape` or `.portrait` based on the device rotation. Thank you 😄

File: Example/Controllers/DisplayController.swift
Comment (ts-leojoseph-r) [reply Example/Controllers/DisplayController.swift:56]: Sure, I will add it in the documentation. I just noticed now that if the host app didn't override `AVPlayerViewController - > override open var supportedInterfaceOrientations` then the orientation lock is not working. This is because the Video player opened from the mini-app. I guess the host app need to implement this for sure, I will update all these things in the documentation once the PR is merged.

File: MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift
Comment (ts-leojoseph-r) [reply MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift:188]: I don't think we need this in iOS. We have implemented it differently in iOS due to the delegation pattern for orientation.

`func setScreenOrientation(requestParam: RequestParameters?, callbackId: String)` will only lock/set the orientation for the mini-app/video and the Host app must use `self.miniAppDisplayDelegate?.getSupportedOrientation()` in DisplayController or any viewcontroller which they are displaying Mini-app view.
And `self.miniAppDisplayDelegate?.getSupportedOrientation()` i.e MiniAppDisplayProtocol will retrieve the orientation supported for that mini-app.

File: Example/Controllers/DisplayController.swift
Comment (ts-leojoseph-r) [reply Example/Controllers/DisplayController.swift:3]: Sadly yes, I don't like it either.
And if we don't use this, then the orientation lock is not working for the video player. Funny thing is, even if the app supports only portrait mode, the video player can rotate to all orientations(without this extension). So having this code in place helps us to lock the orientation.

**Review by corycaywood — APPROVED** [review]
Looks good 👍

File: MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift
Comment (corycaywood) [inline MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift:199]: Sorry, I missed something here. We need to send a `success` callback to the mini app. The Android SDK sends success after then orientation was successfully set.

It looks like there isn't a way to detect if `attemptRotationToDeviceOrientation` was successful? So maybe it's okay just to send success immediately after calling `attemptRotationToDeviceOrientation`?

Comment (ts-leojoseph-r) [reply MiniApp/Classes/JavascriptBridge/MiniAppScriptMessageHandler.swift:199]: Oh, sorry. I thought Android isn't sending any success response, I looked at few android comments. No probs, I will make changes in another PR 👍

**Pull request merged**
